course_info/tests/jasmine/controllers/PeopleControllerSpec.js

```diff
@@ -128,19 +160,540 @@ describe('Unit testing PeopleController', function() {
+                    {active: 0, prime_role_indicator: 'N'},
+                    {active: 0, prime_role_indicator: ''},
+                ];
+
+                roles.sort(scope.compareRoles);
+                expect(roles).toEqual(afterSorting);
+            });
+        });
+
+        describe('disableAddUserButton', function() {
+            // the state of the add user button depends on the contents of
+            // the search field, on whether a lookup has returned multiple
+            // results, and on whether any of those results have been selected
+            it('should disable if a search is in progress', function() {
+                scope.searchInProgress = true;
+                expect(scope.disableAddUserButton()).toBe(true);
+            });
+            it('should enable if the search had multiple results and one is selected',
+               function() {
+                   scope.searchResults = [{}]; // contents don't matter, only length
+                   scope.selectedResult = {id: 123};
+                   expect(scope.disableAddUserButton()).toBe(false);
+               }
+            );
+            it('should disable if the search had multiple results and none are selected',
+               function() {
+                   scope.searchResults = [{}]; // contents don't matter, only length
+                   scope.selectedResult = {id: null};
+                   expect(scope.disableAddUserButton()).toBe(true);
+               }
+            );
+            it('should enable if a search term has been entered and there are no results',
+               function() {
+                   scope.searchTerm = 'bob';
+                   expect(scope.disableAddUserButton()).toBe(false);
+               }
+            );
+            it('should disable if there is no search term or results', function() {
+                expect(scope.disableAddUserButton()).toBe(true);
+            });
+        });
+
+        describe('filterResults', function() {
+            // NOTE: relies on compareRoles() working properly.  mocking
+            //       its results wasn't worth it.
+            it('should not blow up on empty input', function() {
+                expect(scope.filterResults([])).toEqual([]);
+            });
+
+            it('should return one role per univ_id', function() {
+                var searchResults = [
+                    {univ_id: 123, active: 0, prime_role_indicator: ''},
+                    {univ_id: 123, active: 1, prime_role_indicator: 'Y'},
+                    {univ_id: 456, active: 0, prime_role_indicator: 'Y'},
+                    {univ_id: 456, active: 1, prime_role_indicator: ''},
+                    {univ_id: 789, active: 0, prime_role_indicator: 'Y'},
+                ];
+                var filtered = scope.filterResults(searchResults);
+                var uniq = {};
+                filtered.forEach(function(r) { uniq[r.univ_id] = true; });
```

Comment:
Won't this just overwrite an existing key, so there will only ever be one value per univ_id anyway? Maybe we should be doing `uniq[r.univ_id] = (r.univ_id in uniq ? uniq[r.univ_id]++ : 1)` or something like that, and checking both `.length()` and that all values are `1`?